### ISSUE TITLE
Redis output ignores default output filter --output-filter="success = 1 && repeat = 0"

### ISSUE THREAD

**Issue by eplewis89** [zmap/zmap#564]
**(see my last comment, this is the path that I took discovering this issue)**

During a zmap scan, I'm writing results to redis as resdis-csv format, and my redis queue is gaining a considerable amount more results than what is being reported in the zmap output.

For example, the 36.0.0.0/8 subnet at a 200/second rate is showing only 23 responses (after 10 minutes), but my redis queue has a factor of 100x the responses in the subnet.

I'm not sure if this is a bug with Redis, or some weird threading issue, but I'm running a second scan along side of the redis scan, outputting results to a file. The second scan is gathering results at a much slower rate even though the rate limit is set to 200.

What info do you need from me to look into this?

**Comment by eplewis89** [comment]
More info, two scans are currently running, port 2223 on the 36.0.0.0/8 CIDR

First scan (the "redis" scan), the command is: "36.0.0.0/8 -p 2223 --output-module=redis-csv --output-args=tcp://x.x.x.x:6379/testresults:2223 -r 200 -f saddr,saddr-raw,sport,timestamp-str"

Second scan (the "file" scan), the command is: "36.0.0.0/8 -p 2223 -o /zmap-results/2223/36.0.0.0-subnet-8.txt -r 200 -f saddr,daddr,sport,dport,seqnum,acknum,timestamp-str"

In the first scan, at 1 minute and 13k packets sent, the redis queue has 1000 items in it, but the zmap output says "1:04 0% (23h left); send: 12671 197 p/s (197 p/s avg); recv: 5 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.04%"

In the second scan, at the 1 minute mark, the zmap output says "1:04 0% (23h left); send: 12803 199 p/s (199 p/s avg); recv: 2 0 p/s (0 p/s avg); drops: 0 p/s (0 p/s avg); hitrate: 0.02%" and the file has 2 results in it.

This is weird right?

**Comment by eplewis89** [comment]
Ok I think I found the issue. In the scan that outputs to file, inherently, the output filter is set to the following: --output-filter="success = 1 && repeat = 0"

Somehow the redis module is ignoring this flag and writing IP addresses out (I didn't look to see whether they were duplicates or not).

**Comment by dadrian** [comment]
The confusiong around when the output-filter does what should now be fixed with #668. However, we also removed the Redis output module, as it has been buggy and is likely not memory safe. Users are better off piping data into Redis using some external scripts in conjunction with ztee, e.g.

```
zmap -p 80 --output-module=csv | ztee --raw | python add_to_my_redis_cluster.py
```
